lib/mix/lib/mix/rebar.ex

```diff
@@ -62,108 +62,6 @@ defmodule Mix.Rebar do
   end
 
   @doc """
-  Merges a rebar3 parent config with a child config.
-  """
-  # From https://github.com/rebar/rebar3/blob/b1da2ec0674df89599564252734bd4d794436425/src/rebar_opts.erl#L103
-  def merge_config(old, new) do
```

Comment:
And apparently we don't need to worry about the parent to child case at all because those projects cannot be used as dependencies.